### PR TITLE
feat(module/asg): Introduce Instance Refresh for ASG module

### PR DESCRIPTION
**Pull request by sebastianczech**
## Description

PR introduces new feature for ASG module - [instance refresh](https://aws.amazon.com/blogs/compute/introducing-instance-refresh-for-ec2-auto-scaling/), which [can be customized by allowing to change every possible attribute for instance refresh block](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#instance_refresh).

## Motivation and Context

[#416](https://github.com/PaloAltoNetworks/terraform-aws-vmseries-modules/issues/416) 

## How Has This Been Tested?

Code was tested by extending example `isolated_design_autoscale` and deploying it in the lab.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by sebastianczech** [comment]
/idempotence paths="examples/centralized_design_autoscale examples/combined_design_autoscale examples/isolated_design_autoscale"

File: examples/centralized_design_autoscale/example.tfvars [hunk 1]
```diff
@@ -576,6 +576,23 @@ vmseries_asgs = {
       }
     }
 
+    launch_template_version = "1"
```
Comment (pimielowski) [inline examples/centralized_design_autoscale/example.tfvars:579]: If there will be a change in launch template we will be still using version 1 not the latest one with the changes.

**Comment by sebastianczech** [comment]
/idempotence paths="examples/centralized_design_autoscale examples/combined_design_autoscale examples/isolated_design_autoscale" tf_version="1.7"

File: examples/centralized_design_autoscale/example.tfvars (hunk 1 above)
Comment (pimielowski) [reply examples/centralized_design_autoscale/example.tfvars:579]: Ok, thanks for the clarification, I was not aware that this is an issue. This is not explicit example for the instance refresh, I would propose to leave like this and give an information about it like a comment or so, to clarify it, are you ok with this approach?

Comment (sebastianczech) [reply examples/centralized_design_autoscale/example.tfvars]: Ok, done.

**Comment by sebastianczech** [comment]
/idempotence paths="examples/centralized_design_autoscale examples/combined_design_autoscale examples/isolated_design_autoscale" tf_version="1.6"

File: examples/centralized_design_autoscale/variables.tf [hunk 2]
```diff
@@ -278,6 +278,8 @@ variable "vmseries_asgs" {
     - `statistic`: statistic of the metric. Valid values: Average, Maximum, Minimum, SampleCount, Sum
     - `cloudwatch_namespace`: name of CloudWatch namespace, where metrics are available (it should be the same as namespace configured in VM-Series plugin in PAN-OS)
     - `tags`: tags configured for dynamic scaling policy
+  - `launch_template_version`: launch template version to use to launch instances
+  - `instance_refresh`: instance refresh for ASG defined by several attributes (please README for module `asg` for more details)
```
Comment (migara) [inline examples/centralized_design_autoscale/variables.tf:282]:
```suggestion
  - `instance_refresh`: instance refresh for ASG defined by several attributes (please see README for module `asg` for more details)
```

**Review by pimielowski — APPROVED** [review]

**Pull request merged**